rust/src/builders/witness_builder.rs

```diff
@@ -303,6 +303,54 @@ impl TransactionWitnessSetBuilder {
         };
     }
 
+    fn add_vkeys(&mut self, vkeys: &Vec<Vkey>) {
+        let fake_sig = fake_raw_key_sig(0);
+        for vkey in vkeys {
+            let fake_vkey_witness = Vkeywitness::new(&vkey, &fake_sig);
+            self.add_vkey(&fake_vkey_witness);
+        }
+    }
+
+    fn add_vkeys_by_num(&mut self, num: usize) {
+        let vkeys: Vec<Vkey> = (0..num).into_iter().map(|_| Vkey::new(&fake_raw_key_public(0))).collect();
```

Comment:
If you're always using the same fake raw key, it will just get deduplicated so it won't properly count towards the tx fee like you want it to.
You could fix this by using the iterator index instead of hardcoding `0`, but this still won't work if `add_vkeys_by_num` gets called twice for the same transaction

Additionally, right now this has a chance of double-counting witnesses. Imagine a transaction that contains 2 Plutus scripts both which have the same required signers. This code would double the amount of fake witnesses added even though a single copy would be enough

I think the proper fix to this would be to: instead of calling this function in multiple places, instead:
1. deduplcate the ED25519 keys from missing signers
    1. easy version: deduplicate missing signers only.
    2. More complicated version I don't think we want: deduplicate missing signers and also remove any missing signer that was added somewhere else as a `vkeys`. Presumably the case this happens is low and it's kind of a user error, so only doing (1) is good enough I think
3. Call this function setting `num` as `deduplicated.len()` + the sum of all the `NativeScriptWitnessInfoKind::Count(num)`